draft-bbs-signatures.md

```diff
@@ -680,38 +680,56 @@ Procedure:
 
 # Utility Operations
 
-## CreateGenerators
+## Generator point computation
 
-The CreateGenerators operation defines how to create a set of generators that form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as Sign, Verify, ProofGen and ProofVerify.
+The `create_generators` operation defines how to create a set of generators that form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as Sign, Verify, ProofGen and ProofVerify. It takes one input, the number of generator points to create, which is determined in part by the number of selectively-revealed signed messages.
+
+As an optimization, implementations MAY cache the result of `create_generators` for a specific `generator_seed` (determined by the ciphersuite) and `count`. The values `c` and `v` MAY also be cached in order to efficiently extend a existing list of generator points.
 
 ```
-generators = CreateGenerators(dst, message_generator_seed, length);
+generators = create_generators(count)
 
 Inputs:
 
-- dst (REQUIRED), octet string. Domain Separation Tag.
-- message_generator_seed (REQUIRED), octet string.
-- length (REQUIRED), unsigned integer. Number of generators to create from the seed and dst.
+- count (REQUIRED), unsigned integer. Number of generator points to create.
+
+Parameters:
+
+- generator_seed, octet string. A seed value selected by the ciphersuite.
+- hash_to_curve_g1. A hash-to-curve suite selected by the ciphersuite.
+
+Definitions:
+
+- expand_message is the corresponding operation associated with the ciphersuite's hash_to_curve_g1 parameter.
+- seed_dst is the octet string representing "BBS-SIG-GENERATOR-SEED" in the ASCII character encoding.
+- hash_output_bytes is the integer given by ceil((log2(r) + k) / 8).
```

Comment:
We don't mod something here so i don't see the reason for this suggestion. This will be the seed that will go into hash_to_curve which can be whatever length we want wright?? Or are we going to use that value, just so we will not introduce an extra variable?? It still seems a little extensive though, why not use `ceil(log2(r)/8)` for example??

Reply:
And actually, thinking more about it, does this method "require" expand_message?? Wouldn't just `hash` do, where if `hash` is an xof, we will specify the output length??

Reply:
`hash_output_bytes` would likely be replaced by `expand_length` in future, but that's in another PR. I didn't want to reference `xof_no_bytes` here.

Reply:
> And actually, thinking more about it, does this method "require" expand_message?? Wouldn't just hash do, where if hash is an xof, we will specify the output length??

It could but that would limit `v` to the hash's output size for fixed output hashes. I prefer using `expand_message` here because it's a primitive that is guaranteed to be available and should be hard to misuse.

Reply:
Ok. Thats a good point and i think that you are right in that expand_message will be just as available in practice.